receiver/mongodbreceiver/scraper_test.go

```diff
@@ -288,7 +288,12 @@ func TestScraperScrape(t *testing.T) {
 			scraperCfg.MetricsBuilderConfig.Metrics.MongodbHealth.Enabled = true
 
 			scraper := newMongodbScraper(receivertest.NewNopCreateSettings(), scraperCfg)
-			scraper.client = tc.setupMockClient(t)
+
+			mc := tc.setupMockClient(t)
+			if mc != nil {
+				scraper.client = mc
+			}
```

Comment:
Does this change anything? 

`newMongodbScraper` doesn't initialize a client, so it starts out as nil. Then we check if `tc.setupMockClient(t)` returned nil so that we don't override nil with nil.

Reply:
That was my first thought, after a nil pointer error, I realized that those nil are not the same 
```go
			mc := tc.setupMockClient(t)
			// if mc != nil {
			fmt.Printf("initialize value: %t\n", scraper.client)
			scraper.client = mc
			fmt.Printf("modified value: %t\n", scraper.client)
			// }
```
```sh
initialize value: %!t(<nil>)
modified value: %!t(*mongodbreceiver.fakeClient=<nil>)
```